qiita_db/artifact.py

```diff
@@ -269,107 +284,163 @@ def create(cls, filepaths, artifact_type, name=None, prep_template=None,
             raise qdb.exceptions.QiitaDBArtifactCreationError(
                 "at least one filepath is required.")
 
-        # Parents or prep template must be provided, but not both
-        if parents and prep_template:
+        # Check that the combination of parameters is correct
+        counts = Counter([bool(parents or processing_parameters),
```

Comment:
Good call, I think when I started writing this I had cases for the counts but it end up evolving and at the end I don't need it - changed to a simple addition of ints (although with a few casts...)